src/network/networkd-link.c

```diff
@@ -1510,7 +1551,20 @@ static int link_up(Link *link) {
                         return log_link_error_errno(link, r, "Could not set MAC address: %m");
         }
 
+        /* If IPv6 not configured (no static IPv6 address and neither DHCPv6 nor IPv6LL is enabled)
+           for this interface then disable IPv6 else enable it. */
+        (void) link_enable_ipv6(link);
+
         if (link->network->mtu) {
+                /* IPv6 protocol requires a minimum MTU of IPV6_MTU_MIN(1280) bytes
+                   on the interface. Bump up MTU bytes to IPV6_MTU_MIN. */
+                if (link_ipv6_enabled(link) && link->network->mtu < IPV6_MIN_MTU) {
+
+                        log_link_warning_errno(link, r, "Bumping MTU to 1280, as IPv6 is requested and requires a minimum MTU of 1280 bytes: %m");
```

Comment:
Passing "r" here is not right. Please use log_link_warning() here, (and not log_link_warning()) and don't pass "r"...